### PR TITLE
Update modals to align with PatternFly standard

### LINKED ISSUE #2661: Improve modal consistency
* use of modal-header
  * use of modal-title
* background color
* button alignment (right) and button order (secondary then primary)
* close by clicking background
  * uibModal --> configurable, default is true
  * overlay-panel --> false

PatternFly modal:
screen shot 2018-01-15 at 11 50 27 am

### PR DESCRIPTION
**Pull request by rhamilto**
And set modal config globally:
  * animation: true
  * backdrop: 'static' // so the default behavior is changed to not allow clicking the backdrop to close the modal

Requires https://github.com/openshift/origin-web-catalog/pull/628

Fixes #2661

attn: @openshift/team-ux-review 

screen shot 2018-01-26 at 3 41 11 pm
screen shot 2018-01-26 at 3 50 37 pm
screen shot 2018-01-26 at 3 50 16 pm
screen shot 2018-01-26 at 3 50 21 pm
screen shot 2018-01-29 at 1 38 25 pm
screen shot 2018-01-29 at 1 39 43 pm
screen shot 2018-01-29 at 1 42 51 pm
screen shot 2018-01-29 at 1 44 48 pm
screen shot 2018-01-29 at 1 46 41 pm
screen shot 2018-01-29 at 1 56 15 pm
screen shot 2018-01-29 at 2 01 57 pm
screen shot 2018-01-29 at 2 03 10 pm
screen shot 2018-01-29 at 4 34 01 pm
screen shot 2018-01-29 at 2 03 42 pm
screen shot 2018-01-29 at 2 05 19 pm
screen shot 2018-01-29 at 2 07 24 pm
screen shot 2018-01-29 at 2 08 43 pm
screen shot 2018-01-29 at 4 16 30 pm
screen shot 2018-01-29 at 2 23 04 pm
screen shot 2018-01-29 at 2 23 14 pm
screen shot 2018-01-29 at 2 45 49 pm
screen shot 2018-01-29 at 2 46 01 pm
screen shot 2018-01-29 at 4 16 54 pm
screen shot 2018-01-30 at 11 32 02 am

### CODE REVIEW

**Comment by spadgett** [comment]
I think the compute resources and Jenkinsfile dialogs were to have "Close" buttons instead of "OK"

This looks so much better.

**Comment by rhamilto** [comment]
> I think the compute resources and Jenkinsfile dialogs were to have "Close" buttons instead of "OK"

Changed "OK" buttons to "Close"

File: app/scripts/controllers/edit/deploymentConfig.js [hunk 1]
```diff
@@ -227,14 +227,13 @@ angular.module('openshiftConsole')
...
       var modalInstance = $uibModal.open({
-        animation: true,
         templateUrl: 'views/modals/confirm.html',
         controller: 'ConfirmModalController',
         resolve: {
           modalConfig: function() {
             return {
               alerts: $scope.alerts,
-              message: "Some of your existing " + $scope.originalStrategy.toLowerCase() + " strategy parameters can be used for the " + $scope.strategyData.type.toLowerCase() + " strategy. Keep parameters?",
+              title: "Some of your existing " + $scope.originalStrategy.toLowerCase() + " strategy parameters can be used for the " + $scope.strategyData.type.toLowerCase() + " strategy. Keep parameters?",
```
Comment (spadgett) [inline app/scripts/controllers/edit/deploymentConfig.js]: This seems like a lot of text to go in the titlebar

File: app/styles/_add-config-to-application.less [hunk 2]
```diff
@@ -26,37 +23,6 @@
     }
   }
 
-  .dialog-title {
```
Comment (spadgett) [inline app/styles/_add-config-to-application.less]: nice to see so much CSS removed

**Comment by ncameronbritt** [comment]
This is a big improvement! There are a couple cases where the title bar is inconsistent and I'm not sure that it needs to be:

For the create image secret, could we pull the "Learn more" link out of the title bar? 
https://user-images.githubusercontent.com/18057044/35409339-e1c300d0-01df-11e8-98e3-fccade6bc05b.png

And for the debug container can we bring the info about the specific containers out of the title bar?
https://user-images.githubusercontent.com/18057044/35409340-e1d33306-01df-11e8-86f4-fb94565cc077.png

**Comment by spadgett** [comment]
@ahardin-rh FYI, we're updating our modal styles in 3.9 if we have any screenshots in the docs.

**Comment by rhamilto** [comment]
> This is a big improvement! There are a couple cases where the title bar is inconsistent and I'm not sure that it needs to be:

IMO, these items are better in the title bar because they maintain context with the title and reduce the amount of vertical space being used in the modal.  But I can be convinced otherwise.  Do you feel strongly?

**Comment by rhamilto** [comment]
Given the large number of files changed, I'd love to get this in as is and address any changes in a follow on.  Any objections?

**Comment by spadgett** [comment]
> Given the large number of files changed, I'd love to get this in as is and address any changes in a follow on. Any objections?

OK by me. This is a clear improvement.

**Pull request merged**